pkg/sql/schemachanger/scplan/internal/rules/current/dep_add_constraint.go

```diff
@@ -47,4 +47,20 @@ func init() {
 			}
 		},
 	)
+
+	// Constraint name should be assigned right before it becomes visible, otherwise
+	// we won't have the correct message inside errors.
+	registerDepRule(
+		"simple constraint visible before name",
+		scgraph.SameStagePrecedence,
+		"simple-constraint", "constraint-name",
+		func(from, to NodeVars) rel.Clauses {
+			return rel.Clauses{
+				from.TypeFilter(rulesVersionKey, isNonIndexBackedConstraint),
```

Comment:
I think we should swap `from` and `to` here? the name should be public before the constraint becomes write-only. Maybe it's fine to leave it as is since it's in same stage, but semantically I think we should swap.

Reply:
This is a weird chicken and egg problem, we need the constraint to be added first before the name can be set. The same stage makes sure the two are done at the same time. Swapping it the name would be set before the constraint even exists :(